src/main/java/ch/njol/skript/expressions/ExprTransform.java

```diff
@@ -0,0 +1,168 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Keywords;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.InputSource;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.parser.ParserInstance;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.Pair;
+import com.google.common.collect.Iterators;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+import org.skriptlang.skript.lang.converter.Converters;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+@Name("Transformed List")
+@Description({
+	"Transforms (or 'maps') a list's values using a given expression. This is akin to looping over the list and getting " +
+	"a modified version of each value.",
+	"Indices cannot be retained with this expression. To retain indices, see the transform effect."
+})
+@Examples({
+	"set {_a::*} to (1, 2, and 3) transformed using (input * 2 - 1, input * 2)",
+	"# {_a::*} is now 1, 2, 3, 4, 5, and 6",
+	"",
+	"# get a list of the sizes of all clans without manually looping",
+	"set {_clan-sizes::*} to indices of {clans::*} transformed using [{clans::%input%::size}]",
+})
+@Since("INSERT VERSION")
+@Keywords("input")
+public class ExprTransform extends SimpleExpression<Object> implements InputSource {
+
+	static {
+		Skript.registerExpression(ExprTransform.class, Object.class, ExpressionType.PATTERN_MATCHES_EVERYTHING,
+				"%~objects% (transformed|mapped) (using|with) \\[<.+>\\]",
+				"%~objects% (transformed|mapped) (using|with) \\(<.+>\\)"
```

Comment:
```suggestion
				"%objects% (transformed|mapped) (using|with) \\[<.+>\\]",
				"%objects% (transformed|mapped) (using|with) \\(<.+>\\)"
```
As discussed on Discord, this can be removed for the expression as it returns a new list (and something like `1 mapped using input + 1`, while weird, has no reason not to work).